### PR TITLE
LPS-178066 CSP part 2

### PR DESCRIPTION
**Pull request by izaera**
This supersedes https://github.com/liferay-frontend/liferay-portal/pull/3587 after part 1 has been merged to master

### CODE REVIEW

**Comment by izaera** [comment]
ci:stop

**Comment by izaera** [comment]
ci:test:sf

**Comment by izaera** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:test:relevant

**Comment by JerryAdrianoo** [comment]
Hey, @izaera  I found two issues possibly related to your changes; see below.
If you're sure it's not related to your changes, we need to run ci:relevant again and see if it's an inconsistent result or not.


```
     [exec] com.liferay.portal.security.membership.policy.site.test.SiteMembershipPolicyRolesTest > testPropagateWhenAssigningRolesToUser FAILED
     [exec]     com.liferay.portal.kernel.exception.UserPasswordException$MustNotBeChangedYet: Password for user 46577 must not be changed until Wed Sep 20 09:15:16 GMT 2023
     [exec]     	at com.liferay.portal.security.pwd.PasswordPolicyToolkit.validate(PasswordPolicyToolkit.java:183)
     [exec]     	at com.liferay.portal.security.pwd.PwdToolkitUtil.validate(PwdToolkitUtil.java:40)
     [exec]     	at com.liferay.portal.service.impl.UserLocalServiceImpl.validatePassword(UserLocalServiceImpl.java:6988)
     [exec]     	at com.liferay.portal.service.impl.UserLocalServiceImpl.updatePassword(UserLocalServiceImpl.java:4948)
```


and 



```
********     [exec] 
     [exec]     Loading jar:file:/opt/dev/projects/github/liferay-portal/bundles/tomcat-9.0.80/webapps/ROOT/WEB-INF/shielded-container-lib/portal-impl.jar!/system.properties
     [exec]     Loading file:/opt/dev/projects/github/liferay-portal/bundles/tomcat-9.0.80/webapps/ROOT/WEB-INF/classes/system-ext.properties
     [exec] 
     [exec] com.liferay.portal.security.content.security.policy.test.ContentSecurityPolicyFilterTest > testProcessFilter STARTED
     [exec] 
     [exec] com.liferay.portal.security.content.security.policy.test.ContentSecurityPolicyFilterTest > testProcessFilter FAILED
     [exec]     java.lang.AssertionError
     [exec]     	at org.junit.Assert.fail(Assert.java:87)
     [exec]     	at org.junit.Assert.assertTrue(Assert.java:42)
     [exec]     	at org.junit.Assert.assertFalse(Assert.java:65)
     [exec]     	at org.junit.Assert.assertFalse(Assert.java:75)
     [exec]     	at com.liferay.portal.security.content.security.policy.test.ContentSecurityPolicyFilterTest.testProcessFilter(ContentSecurityPolicyFilterTest.java:161)
```

**Comment by izaera** [comment]
@JerryAdrianoo thx for the analysis. Here are my conclusions:

The first one works locally:

image

And given that it is totally unrelated to the PR's content, we can assume it was a false positive. However, if you want to run relevant OK, I'm OK.

----

The second one is most probably broken due to the PR, so I'll fix it and push another commit and we should be OK.

**Comment by JerryAdrianoo** [comment]
@izaera Excellent!  I'll run ci:relevant again and see... if the same issue occurs, we can assume it's related.

**Comment by JerryAdrianoo** [comment]
ci:test:relevant

**Comment by JerryAdrianoo** [comment]
Hey, @izaera  the problem I mentioned before is related to your changes because it appears again when I re-executed, appears again. Look at the Jenkins log below, please! The same error.

```
     [exec] com.liferay.portal.security.membership.policy.organization.test.OrganizationMembershipPolicyRolesTest > testUnassignRequiredRolesFromUser FAILED
     [exec]     com.liferay.portal.kernel.exception.UserPasswordException$MustNotBeChangedYet: Password for user 44761 must not be changed until Fri Sep 22 12:52:14 GMT 2023
     [exec]     	at com.liferay.portal.security.pwd.PasswordPolicyToolkit.validate(PasswordPolicyToolkit.java:183)
     [exec]     	at com.liferay.portal.security.pwd.PwdToolkitUtil.validate(PwdToolkitUtil.java:40)
     [exec]     	at com.liferay.portal.service.impl.UserLocalServiceImpl.validatePassword(UserLocalServiceImpl.java:6988)
     [exec]     	at com.liferay.portal.service.impl.UserLocalServiceImpl.updatePassword(UserLocalServiceImpl.java:4948)
     [exec]     	at com.liferay.portal.service.impl.UserLocalServiceImpl.updatePassword(UserLocalServiceImpl.java:4905)
     [exec]     	at com.liferay.portal.service.impl.UserLocalServiceImpl.updateUser(UserLocalServiceImpl.java:5316)
     [exec]     	at sun.reflect.GeneratedMethodAccessor4498.invoke(Unknown Source)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.portal.kernel.bean.ClassLoaderBeanHandler.invo
```

File: modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/ClientExtensionBottomJSPDynamicInclude.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
```
Comment (bryceosterhaus) [inline modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/ClientExtensionBottomJSPDynamicInclude.java:2]: This isn't right. Not sure how it happened, but I remember seeing on slack that someone else also had issues with the copyright updating to 2023, maybe through their editor or something

File: modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java [hunk 2]
```diff
@@ -229,7 +230,7 @@ private String _renderFragmentEntry(
 		FragmentRendererContext fragmentRendererContext, String html,
 		HttpServletRequest httpServletRequest) {
 
-		StringBundler sb = new StringBundler(25);
+		StringBundler sb = new StringBundler(26);
```
Comment (bryceosterhaus) [inline modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java:233]: more of just a noob question here... why did you increment this? and how did you know to increment it?

**Review by bryceosterhaus — APPROVED** [review]
LGTM! Just need to change the copyright before sending to brian, he will reject that change

File: modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java (hunk 2 above)
Comment (izaera) [reply modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java:233]: That is done automatically by SF. I assume it counts the number of strings we are concatenating and uses that to write that number.

File: modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/ClientExtensionBottomJSPDynamicInclude.java (hunk 1 above)
Comment (izaera) [reply modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/ClientExtensionBottomJSPDynamicInclude.java:2]: Mmm, I'm relying on the SF to fix these things and the editor doesn't change anything, AFAIK. Strange...

File: modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java (hunk 2 above)
Comment (bryceosterhaus) [reply modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java:233]: eric-wareheim-mind-blown-gif-by-tim-and-eric

File: modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/ClientExtensionBottomJSPDynamicInclude.java (hunk 1 above)
Comment (bryceosterhaus) [reply modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/ClientExtensionBottomJSPDynamicInclude.java:2]: Here was the past discussion I saw on it https://liferay.slack.com/archives/C1SJ64S9X/p1694676523871839

**Comment by izaera** [comment]
We are going to restructure these PRs, so I'm closing them.

**Pull request closed without merging**